src/subplots.ts

```diff
@@ -2549,6 +2576,107 @@ export class newScatter extends Frame {
   }
 }
 
+export class newGraph2D extends newScatter {
+  public curves: LineSequence[];
+  private curvesIndices: number[][];
+  public showPoints: boolean = false;
+  constructor(
+    data: any,
+    public width: number,
+    public height: number,
+    public buttons_ON: boolean,
+    public X: number,
+    public Y: number,
+    public canvas_id: string,
+    public is_in_multiplot: boolean = false
+    ) {
+      super(data, width, height, buttons_ON, X, Y, canvas_id, is_in_multiplot);
+    }
+
+  protected unpackData(data: any): Map<string, any[]> {
+    const formattedData: {[key: string]: any} = {};
+    this.pointStyles = [];
+    this.curvesIndices = [];
+    this.curves = [];
+    formattedData["elements"] = [];
+    if (data.graphs) {
+      data.graphs.forEach(graph => {
+        this.curves.push(LineSequence.getGraphProperties(graph));
+        const curveIndices = range(formattedData["elements"].length, formattedData["elements"].length + graph.elements.length);
+        this.pointStyles.push(...new Array(curveIndices.length).fill({}));
+        this.curvesIndices.push(curveIndices);
+        this.unpackGraphPointStyle(curveIndices, graph.point_style);
+        formattedData["elements"].push(...graph.elements);
+      })
+    }
+    return super.unpackData(formattedData)
+  }
+
+  private unpackGraphPointStyle(pointsIndices: number[], packedPointStyle: {[key: string]: any}): void {
+    if (packedPointStyle?.shape) pointsIndices.forEach(i => this.pointStyles[i].marker = packedPointStyle.shape);
```

Comment:
If `this.pointStyles[i]` is an Object, these statements should probably be a method of that object

Reply:
Done